src/backend/marsha/bbb/serializers.py

```diff
@@ -111,3 +122,135 @@ def get_lti_url(self, obj):
             self.context["request"],
             reverse("bbb:classroom_lti_view", args=[obj.id]),
         )
+
+
+class ClassroomDocumentSerializer(
+    UploadableFileWithExtensionSerializerMixin, serializers.ModelSerializer
+):
+    """A serializer to display a ClassroomDocument resource."""
+
+    class Meta:  # noqa
+        model = ClassroomDocument
+        fields = (
+            "classroom",
+            "filename",
+            "id",
+            "is_default",
+            "upload_state",
+            "uploaded_on",
+            "url",
+        )
+        read_only_fields = (
+            "classroom",
+            "id",
+            "upload_state",
+            "uploaded_on",
+            "url",
+        )
+
+    url = serializers.SerializerMethodField()
+    # Make sure classroom UUID is converted to a string during serialization
+    classroom = serializers.PrimaryKeyRelatedField(
+        read_only=True, pk_field=serializers.CharField()
+    )
+
+    def create(self, validated_data):
+        """Force the classroom field to the classroom of the JWT Token if any.
+
+        Parameters
+        ----------
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "validated_data" dictionary is returned after modification.
+
+        """
+        resource = self.context["request"].resource
+
+        if not validated_data.get("classroom_id") and resource:
+            validated_data["classroom_id"] = resource.id
+
+        classroom = Classroom.objects.get(id=validated_data["classroom_id"])
+        if classroom.classroom_documents.count() == 0:
+            validated_data["is_default"] = True
+
+        return super().create(validated_data)
+
+    def update(self, instance, validated_data):
+        """Set the default attribute of the others classroom documents to false if
+        set to true for this one.
+
+        Parameters
+        ----------
+        instance : Type[models.ClassroomDocument]
+            The document that we want to update
+
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        Type[models.ClassroomDocument]
+            The updated document
+
+        """
+        if validated_data.get("is_default"):
+            instance.classroom.classroom_documents.exclude(id=instance.id).filter(
+                is_default=True
+            ).update(is_default=False)
+
+        return super().update(instance, validated_data)
```

Comment:
I think this should be called first and once the instance updated then you change `is_default`.